File: 351003/Suchok/publisher/src/main/java/by/liza/app/model/Writer.java
package by.liza.app.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(
        name = "tbl_writer",
        schema = "distcomp",
        uniqueConstraints = @UniqueConstraint(name = "uq_writer_login", columnNames = "login")
)
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(of = "id")
@ToString(exclude = "articles")
public class Writer {

    public enum Role {
        ADMIN, CUSTOMER
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 64)
    private String login;

    @Column(nullable = false, length = 128)
    private String password;

    @Column(nullable = false, length = 64)
    private String firstname;

    @Column(nullable = false, length = 64)
    private String lastname;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 16)
    @Builder.Default
    private Role role = Role.CUSTOMER;

    @OneToMany(
            mappedBy = "writer",
            cascade = CascadeType.ALL,
            orphanRemoval = true,
            fetch = FetchType.LAZY
    )
    @Builder.Default
    private List<Article> articles = new ArrayList<>();
}

File: 351003/Soika/entities-app/publisher/src/main/kotlin/com/example/entitiesapp/model/Writer.kt
package com.example.entitiesapp.model

import jakarta.persistence.*
import com.example.entitiesapp.dto.Role

@Entity
@Table(name = "tbl_writer", schema = "distcomp")
data class Writer(
    override var id: Long? = null,

    @Column(nullable = false, unique = true, length = 64)
    var login: String,

    @Column(nullable = false, length = 128)
    var password: String,

    @Column(nullable = false, length = 64)
    var firstname: String,

    @Column(nullable = false, length = 64)
    var lastname: String,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 16)
    var role: Role = Role.CUSTOMER
) : BaseEntity(id)